builtin/joined_dataset.cc

```diff
@@ -973,53 +985,120 @@ getChildAliases(std::vector<Utf8String> & outAliases) const
 
 BoundFunction
 JoinedDataset::
-overrideFunction(const Utf8String & tableName,
-                 const Utf8String & functionName,
-                 SqlBindingScope & scope) const
-{
-    //cerr << "JoinedDataset function name: " << functionName << " from table: " << tableName << endl;
-    if (functionName == "rowName") {
-
-        if (tableName.empty())
-            return BoundFunction();
-
-        //we do as much "side-checking" as we can at binding time
-
-        JoinedDataset::Itl::JoinSide tableSide = JoinedDataset::Itl::JOIN_SIDE_MAX;
+overrideFunctionFromSide(JoinSide tableSide,
+                         const Utf8String & tableName,
+                         const Utf8String & functionName,
+                         SqlBindingScope & scope) const {
 
-        if (tableName == itl->getTableAlias(JoinedDataset::Itl::JOIN_SIDE_LEFT))
-            tableSide = JoinedDataset::Itl::JOIN_SIDE_LEFT;
-        else if (tableName == itl->getTableAlias(JoinedDataset::Itl::JOIN_SIDE_RIGHT))
-            tableSide = JoinedDataset::Itl::JOIN_SIDE_RIGHT;
-
-        if (tableSide != JoinedDataset::Itl::JOIN_SIDE_MAX)
-        {
-            return {[&, tableSide] (const std::vector<ExpressionValue> & args,
+    ExcAssert(tableSide != JOIN_SIDE_MAX);
+    if (functionName == "rowName") {
+        return {[&, tableSide] (const std::vector<ExpressionValue> & args,
                      const SqlRowScope & scope)
-                { 
+                {
                     auto & row = scope.as<SqlExpressionDatasetScope::RowScope>();
                     return ExpressionValue(itl->getSubRowName(row.row.rowName, tableSide).toUtf8String(), Date::negativeInfinity());
                 },
                 std::make_shared<Utf8StringValueInfo>()
             };
-        }
+    }
+    else if (functionName == "rowHash"){
+        return {[&, tableSide] (const std::vector<ExpressionValue> & args,
+                     const SqlRowScope & scope)
+                {
+                    auto & row = scope.as<SqlExpressionDatasetScope::RowScope>();
+                    return ExpressionValue(itl->getSubRowHash(row.row.rowName, tableSide), Date::negativeInfinity());
+                },
+                std::make_shared<Uint64ValueInfo>()
+            };
+    }
+    else {
+        ExcAssert(functionName == "rowPath");
```

Comment:
But this is an obvious programmer error... if someone calls this method with a function it doesnt support, its because a programmer explictly made it so. The error should be caught before even the change is merged, and the assert here will make it *much* easier to find the source of the problem?